.github/workflows/ubuntu.yml

```diff
@@ -205,29 +211,16 @@ jobs:
           cd ../
           rm -r compiled-planner-${{ matrix.ubuntu-version }}
 
-      - name: setup-dependencies
-        run: |
-          pip3 install pytest tox
-          sudo apt-get update
-          sudo apt-get -y install clang-tidy-8 valgrind zlib1g-dev libgmp3-dev
-          sudo apt-get -y install ${{ matrix.compiler-version }}
-
       - name: test
         env:
+          CC: ${{ matrix.compiler-version.cc }}
+          CXX: ${{ matrix.compiler-version.cxx }}
           DOWNWARD_CPLEX_ROOT: /home/runner/work/downward/lib/ibm/ILOG/CPLEX_Studio129/cplex
           DOWNWARD_SOPLEX_ROOT: /home/runner/work/downward/lib/soplex-3.1.1
           DOWNWARD_COIN_ROOT: /home/runner/work/downward/lib/coin
         run: |
           export PATH="$(pwd):$PATH"  # Add uncrustify and VAL to PATH.
           cd misc/
-          python_version=${{ matrix.python-version }}
-          if [[ $python_version=="3.6" ]]; then
-            tox -e py36,translator,search,style
-          elif [[ $python_version=="3.8" ]]; then
-            tox -e py38,translator,search,style
-          else
-            echo "Unknown Python version";
-            exit 1;
-          fi
+          tox -e py,translator,search,style
```

Comment:
I think the search and style tests should only be run once, not thrice. They can take up quite some time.

Reply:
What do you mean? Are they part of the py test?

Reply:
No, the tox call is excecuted for 18.04-3.6, 20.04-3.6 and 20.04-3.8 if I'm not overlooking anything.

Reply:
Also, we're currently missing the clang-tidy and valgrind tests.

Reply:
Thanks to our quick discussion on discord, I now understand what the py test does. But why do we perform exactly the exit code and autodoc test with different python versions? I think I don't understand what exactly we want to test with different python versions: all of the tox tests? all of the python scripts (i.e., translator and driver)? both? Figuring this out could also potentially help with the artifacts problem.

Reply:
There's some Python code that exists solely for testing the C++ code (search, clang-tidy, valgrind). We only need to run it using a single Python version. The remaining Python code (driver incl. exit codes, autodoc, translator) needs to be tested with multiple Python versions. (So yes, the translator tests should ideally be under the default testenv in tox.ini. This is an oversight that comes from the fact that theses tests used to compare different Python versions instead of the same Python version with different random seeds.)